crypto/keyring/keyring.go

```diff
@@ -927,22 +936,26 @@ func (ks keystore) migrate(version uint32, i keyring.Item, migrated bool) (bool,
 		}
 
 		//5.overwrite the keyring entry with
-		ks.db.Set(keyring.Item{
+		if err := ks.db.Set(keyring.Item{
 			Key:         key,
 			Data:        serializedRecord,
 			Description: "SDK kerying version",
-		})
+		}); err != nil {
+			return migrated, fmt.Errorf("unable to set keyring.Item, err: %w", err)
+		}
 
 		migrated = true
```

Comment:
I introduce bool return type in `checkMigrate()` and `migrate(version uint32, i keyring.Item, migrated bool)`. This return type holds true if a key was successfully migrated.